hazelcast/src/main/java/com/hazelcast/spi/serialization/SerializationService.java

```diff
@@ -0,0 +1,127 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.spi.serialization;
+
+import com.hazelcast.core.ManagedContext;
+import com.hazelcast.core.PartitioningStrategy;
+import com.hazelcast.internal.serialization.PortableContext;
+import com.hazelcast.nio.BufferObjectDataInput;
+import com.hazelcast.nio.BufferObjectDataOutput;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.nio.serialization.PortableReader;
+
+import java.io.IOException;
+import java.nio.ByteOrder;
+
+/**
+ * SPI to serialize user objects to {@link Data} and back to Object
+ * {@link Data} is the internal representation of binary data in hazelcast.
+ */
+public interface SerializationService {
+
+    /**
+     * Serializes an object to a {@link Data}.
+     * <p/>
+     * This method can safely be called with a {@link Data} instance. In that case, that instance is returned.
+     * <p/>
+     * If this method is called with null, null is returned.
+     *
+     * @param obj the object to serialize.
+     * @return the serialized object.
+     * @throws com.hazelcast.nio.serialization.HazelcastSerializationException when serialization fails.
+     */
+    <B extends Data> B toData(Object obj);
+
+    /**
+     * Serializes an object to a {@link Data}.
+     * <p/>
+     * This method can safely be called with a {@link Data} instance. In that case, that instance is returned.
+     * <p/>
+     * If this method is called with null, null is returned.
+     *
+     * @param obj      the object to serialize.
+     * @param strategy strategy is used to calculate partition id of the resulting data see {@link PartitioningStrategy}
+     * @return the serialized object.
+     * @throws com.hazelcast.nio.serialization.HazelcastSerializationException when serialization fails.
+     */
+    <B extends Data> B toData(Object obj, PartitioningStrategy strategy);
+
+    /**
+     * Deserializes an object.
+     * <p/>
+     * This method can safely be called on an object that is already deserialized. In that case, that instance
+     * is returned.
+     * <p/>
+     * If this method is called with null, null is returned.
+     *
+     * @param data the data to deserialize.
+     * @return the deserialized object.
+     * @throws com.hazelcast.nio.serialization.HazelcastSerializationException when deserialization fails.
+     */
+    <T> T toObject(Object data);
+
+    /**
+     * see {@link com.hazelcast.config.Config#setManagedContext(ManagedContext)}
+     *
+     * @return ManagedContext that is set by user in Config
+     */
+    ManagedContext getManagedContext();
+
+    @Deprecated
```

Comment:
Why are these methods added to a new SPI interface but immediately deprecated? That doesn't sound right to me.